syntax = "proto3";

/*
messages between client(caller) and server(auctionhouse)
*/
message Null {};

message receipt {
   bool received = 1;
}

message result {
   float t = 1;
   float subject_bid = 2;
   bool user_win_flag = 3;
   float current_payout = 4;
   float total_winnings = 5;
}

message survey {
   float t = 1;
   float enjoyment = 2;
   float rpe = 3;
}

message treadmill {
   bool state = 1;
}

message testmsg {
   string msg = 1;
}

service auction {
   // caller sends bids to auctionhouse
   rpc testconnection (testmsg) returns (receipt) {};
   rpc call (result) returns (receipt) {};
   rpc question (survey) returns (receipt) {};
   rpc treadmill_message (treadmill) returns (receipt) {};
}